src/log-entry.ts

```diff
@@ -134,12 +135,12 @@ export class Meta {
 /**
  * Represents a complete log entry.
  */
-export class LogEntry {
+export interface LogEntry {
   /** The ID of the request */
-  requestId?: string;
+  requestId?: string | null;
```

Comment:
The null is unnecessary

Reply:
this.sessionId and this.currentRequestId  are set to null in the flow class before being subsequently updated to string, Typescript will not accept that, if there types are not set to string | (or) null

Reply:
Set it to undefined instead of null.